### ISSUE TITLE
Error when trying to export as CSV file in ROOMS

### ISSUE THREAD

**Issue by caripang** [babybunny/rebuildingtogethercaptain#240]
http://rebuildingtogethercaptain.appspot.com/room/captain/export

TypeError at /room/captain/export
sequence item 0: expected string, NoneType found
Request Method:	GET
Request URL:	http://rebuildingtogethercaptain.appspot.com/room/captain/export
Exception Type:	TypeError
Exception Value:	
sequence item 0: expected string, NoneType found
Exception Location:	/base/data/home/apps/s~rebuildingtogethercaptain-hrd/issue233.394576404661038179/room/views.py in CaptainExport, line 559
Python Executable:	/base/
Python Version:	2.5.2
Python Path:	['/base/data/home/apps/s~rebuildingtogethercaptain-hrd/issue233.394576404661038179', '/base/data/home/runtimes/python/python_dist/lib/python25.zip', '/base/data/home/runtimes/python/python_lib/versions/third_party/django-1.0', '/base/data/home/runtimes/python/python_dist/lib/python2.5/', '/base/data/home/runtimes/python/python_dist/lib/python2.5/plat-linux3', '/base/data/home/runtimes/python/python_dist/lib/python2.5/lib-tk', '/base/data/home/runtimes/python/python_dist/lib/python2.5/lib-dynload', '/base/data/home/runtimes/python/python_lib/versions/1', '/base/data/home/apps/s~rebuildingtogethercaptain-hrd/issue233.394576404661038179/']
Server time:	Thu, 9 Feb 2017 16:34:13 +0000
Traceback Switch to copy-and-paste view

/base/data/home/runtimes/python/python_lib/versions/third_party/django-1.0/django/core/handlers/base.py in get_response
                response = callback(request, *callback_args, **callback_kwargs) ...
▶ Local vars
/base/data/home/apps/s~rebuildingtogethercaptain-hrd/issue233.394576404661038179/room/views.py in CaptainExport
    type = '+'.join(set(s.type for s in sc)) ...
▶ Local vars
Request information

GET
No GET data
POST
No POST data
COOKIES
Variable	Value
ACSID	
'~AJKiYcGcwZ44q6pxb1PgzizkdH08dBdFolaENvMompr9jle4a6gnZaZrQzIYRk2bVkZ_WH41KdEECClsfwTlTX_ATD2HWmfhZRCo1ABi3xhkuV10mOvz2umipHXhnj4ejFIg54Y_L2j4NXqCtvg-XfsGEFHq8WQMx1C2D75YypYJ8uv44zloJNRAcoZ5KJK3i1O9Q3uSaowEUxOtgDe5DcMigXoX_8VNgPcWRxxXRmpGCI6P4aj9e-3llfeNhyTEAhPP9y21JYLadGhjRR5xGSctCCoSu9I5yoFjjbydP5E_i4LlTr49bcGDu-i2IDD21HI7OW9-0KwZuZ3Pf6gCoNjQRsnZKU631mVoD8Ww0eiAVu7Xgc3oKBM'
__utmb	
'79689751.3.10.1486657210'
__utmc	
'79689751'
__utmz	
'79689751.1461295397.1.1.utmcsr=(direct)|utmccn=(direct)|utmcmd=(none)'
__utma	
'79689751.38131064.1461295397.1486651269.1486657210.532'
META
Variable	Value
APPENGINE_RUNTIME	
'python'
APPLICATION_ID	
's~rebuildingtogethercaptain-hrd'
AUTH_DOMAIN	
'gmail.com'
CURRENT_MODULE_ID	
'default'
CURRENT_VERSION_ID	
'issue233.394576404661038179'
DATACENTER	
'us2'
DEFAULT_VERSION_HOSTNAME	
'rebuildingtogethercaptain-hrd.appspot.com'
HTTPS	
'off'
HTTP_ACCEPT	
'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8'
HTTP_ACCEPT_LANGUAGE	
'en-US,en;q=0.8'
HTTP_COOKIE	
'ACSID=~AJKiYcGcwZ44q6pxb1PgzizkdH08dBdFolaENvMompr9jle4a6gnZaZrQzIYRk2bVkZ_WH41KdEECClsfwTlTX_ATD2HWmfhZRCo1ABi3xhkuV10mOvz2umipHXhnj4ejFIg54Y_L2j4NXqCtvg-XfsGEFHq8WQMx1C2D75YypYJ8uv44zloJNRAcoZ5KJK3i1O9Q3uSaowEUxOtgDe5DcMigXoX_8VNgPcWRxxXRmpGCI6P4aj9e-3llfeNhyTEAhPP9y21JYLadGhjRR5xGSctCCoSu9I5yoFjjbydP5E_i4LlTr49bcGDu-i2IDD21HI7OW9-0KwZuZ3Pf6gCoNjQRsnZKU631mVoD8Ww0eiAVu7Xgc3oKBM; __utma=79689751.38131064.1461295397.1486651269.1486657210.532; __utmb=79689751.3.10.1486657210; __utmc=79689751; __utmz=79689751.1461295397.1.1.utmcsr=(direct)|utmccn=(direct)|utmcmd=(none)'
HTTP_HOST	
'rebuildingtogethercaptain.appspot.com'
HTTP_REFERER	
'http://rebuildingtogethercaptain.appspot.com/room/captain/list'
HTTP_UPGRADE_INSECURE_REQUESTS	
'1'
HTTP_USER_AGENT	
'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/56.0.2924.87 Safari/537.36'
HTTP_X_APPENGINE_CITY	
'redwood city'
HTTP_X_APPENGINE_CITYLATLONG	
'37.485215,-122.236355'
HTTP_X_APPENGINE_COUNTRY	
'US'
HTTP_X_APPENGINE_REGION	
'ca'
HTTP_X_CLOUD_TRACE_CONTEXT	
'2e7d71821a49672049157458c1ec49ca/8379785783669091224;o=1'
INSTANCE_ID	
'00c61b117c88135278042bf804c48e41f0159fbdcd7ce64214596d77c33a37c63351f7519b6bc8'
PATH_INFO	
u'/room/captain/export'
PATH_TRANSLATED	
'/base/data/home/apps/s~rebuildingtogethercaptain-hrd/issue233.394576404661038179/main.py'
QUERY_STRING	
''
REMOTE_ADDR	
'76.14.79.217'
REQUEST_ID_HASH	
'74DD9BEC'
REQUEST_LOG_ID	
'589c9a0100ff01109874dd9bec0001737e72656275696c64696e67746f6765746865726361707461696e2d68726400016973737565323333000100'
REQUEST_METHOD	
'GET'
SCRIPT_NAME	
u''
SERVER_NAME	
'rebuildingtogethercaptain.appspot.com'
SERVER_PORT	
'80'
SERVER_PROTOCOL	
'HTTP/1.1'
SERVER_SOFTWARE	
'Google App Engine/1.9.48'
TZ	
'UTC'
USER	
'apphosting'
USER_EMAIL	
'cari@rebuildingtogetherpeninsula.org'
USER_ID	
'100635658095976636722'
USER_IS_ADMIN	
'1'
USER_NICKNAME	
'cari'
USER_ORGANIZATION	
'rebuildingtogetherpeninsula.org'
wsgi.errors	
<cStringIO.StringO object at 0xd68f840430e62db8>
wsgi.input	
<cStringIO.StringI object at 0xd68f8404328e27a8>
wsgi.multiprocess	
False
wsgi.multithread	
False
wsgi.run_once	
True
wsgi.url_scheme	
'http'
wsgi.version	
(1, 0)
Settings
Using settings module settings
Setting	Value
ABSOLUTE_URL_OVERRIDES	
{}
ADMINS	
(('Luke Stone', 'babybunny@gmail.com'),)
ADMIN_FOR	
()
ADMIN_MEDIA_PREFIX	
'/media/'
ALLOWED_INCLUDE_ROOTS	
()
APPEND_SLASH	
True
AUTHENTICATION_BACKENDS	
('django.contrib.auth.backends.ModelBackend',)
BANNED_IPS	
()
CACHE_BACKEND	
'locmem://'
CACHE_MIDDLEWARE_KEY_PREFIX	
''
CACHE_MIDDLEWARE_SECONDS	
600
COMMENTS_ALLOW_PROFANITIES	
False
COMMENTS_BANNED_USERS_GROUP	
None
COMMENTS_FIRST_FEW	
0
COMMENTS_MODERATORS_GROUP	
None
COMMENTS_SKETCHY_USERS_GROUP	
None
DATABASE_ENGINE	
'appengine'
DATABASE_HOST	
''
DATABASE_NAME	
''
DATABASE_OPTIONS	
{}
DATABASE_PASSWORD	
'********************'
DATABASE_PORT	
''
DATABASE_USER	
''
DATETIME_FORMAT	
'N j, Y, P'
DATE_FORMAT	
'N j, Y'
DEBUG	
True
DEBUG_PROPAGATE_EXCEPTIONS	
False
DEFAULT_CHARSET	
'utf-8'
DEFAULT_CONTENT_TYPE	
'text/html'
DEFAULT_FILE_STORAGE	
'django.core.files.storage.FileSystemStorage'
DEFAULT_FROM_EMAIL	
'webmaster@localhost'
DEFAULT_INDEX_TABLESPACE	
''
DEFAULT_TABLESPACE	
''
DISALLOWED_USER_AGENTS	
()
EMAIL_HOST	
''
EMAIL_HOST_PASSWORD	
'********************'
EMAIL_HOST_USER	
''
EMAIL_PORT	
25
EMAIL_SUBJECT_PREFIX	
'[Django] '
EMAIL_USE_TLS	
False
FILE_CHARSET	
'utf-8'
FILE_UPLOAD_HANDLERS	
('django.core.files.uploadhandler.MemoryFileUploadHandler', 'django.core.files.uploadhandler.TemporaryFileUploadHandler')
FILE_UPLOAD_MAX_MEMORY_SIZE	
2621440
FILE_UPLOAD_PERMISSIONS	
None
FILE_UPLOAD_TEMP_DIR	
None
FIXTURE_DIRS	
()
FORCE_SCRIPT_NAME	
None
IGNORABLE_404_ENDS	
('mail.pl', 'mailform.pl', 'mail.cgi', 'mailform.cgi', 'favicon.ico', '.php')
IGNORABLE_404_STARTS	
('/cgi-bin/', '/_vti_bin', '/_vti_inf')
INSTALLED_APPS	
('appengine_django', 'gaeunit', 'django.contrib.humanize')
INTERNAL_IPS	
()
JING_PATH	
'/usr/bin/jing'
LANGUAGES	
(('ar', 'Arabic'), ('bn', 'Bengali'), ('bg', 'Bulgarian'), ('ca', 'Catalan'), ('cs', 'Czech'), ('cy', 'Welsh'), ('da', 'Danish'), ('de', 'German'), ('el', 'Greek'), ('en', 'English'), ('es', 'Spanish'), ('et', 'Estonian'), ('es-ar', 'Argentinean Spanish'), ('eu', 'Basque'), ('fa', 'Persian'), ('fi', 'Finnish'), ('fr', 'French'), ('ga', 'Irish'), ('gl', 'Galician'), ('hu', 'Hungarian'), ('he', 'Hebrew'), ('hi', 'Hindi'), ('hr', 'Croatian'), ('is', 'Icelandic'), ('it', 'Italian'), ('ja', 'Japanese'), ('ka', 'Georgian'), ('ko', 'Korean'), ('km', 'Khmer'), ('kn', 'Kannada'), ('lv', 'Latvian'), ('lt', 'Lithuanian'), ('mk', 'Macedonian'), ('nl', 'Dutch'), ('no', 'Norwegian'), ('pl', 'Polish'), ('pt', 'Portuguese'), ('pt-br', 'Brazilian Portuguese'), ('ro', 'Romanian'), ('ru', 'Russian'), ('sk', 'Slovak'), ('sl', 'Slovenian'), ('sr', 'Serbian'), ('sv', 'Swedish'), ('ta', 'Tamil'), ('te', 'Telugu'), ('th', 'Thai'), ('tr', 'Turkish'), ('uk', 'Ukrainian'), ('zh-cn', 'Simplified Chinese'), ('zh-tw', 'Traditional Chinese'))
LANGUAGES_BIDI	
('he', 'ar', 'fa')
LANGUAGE_CODE	
'en-us'
LANGUAGE_COOKIE_NAME	
'django_language'
LOCALE_PATHS	
()
LOGIN_REDIRECT_URL	
'/accounts/profile/'
LOGIN_URL	
'/accounts/login/'
LOGOUT_URL	
'/accounts/logout/'
MANAGERS	
(('Luke Stone', 'babybunny@gmail.com'),)
MEDIA_ROOT	
''
MEDIA_URL	
''
MIDDLEWARE_CLASSES	
('django.middleware.common.CommonMiddleware',)
MONTH_DAY_FORMAT	
'F j'
PASSWORD_RESET_TIMEOUT_DAYS	
'********************'
PREPEND_WWW	
False
PROFANITIES_LIST	
'********************'
ROOT_PATH	
'/base/data/home/apps/s~rebuildingtogethercaptain-hrd/issue233.394576404661038179'
ROOT_URLCONF	
'urls'
SECRET_KEY	
'********************'
SEND_BROKEN_LINK_EMAILS	
False
SERIALIZATION_MODULES	
{'xml': 'appengine_django.serializer.xml'}
SERVER_EMAIL	
'root@localhost'
SESSION_COOKIE_AGE	
1209600
SESSION_COOKIE_DOMAIN	
None
SESSION_COOKIE_NAME	
'sessionid'
SESSION_COOKIE_PATH	
'/'
SESSION_COOKIE_SECURE	
False
SESSION_ENGINE	
'django.contrib.sessions.backends.db'
SESSION_EXPIRE_AT_BROWSER_CLOSE	
False
SESSION_FILE_PATH	
None
SESSION_SAVE_EVERY_REQUEST	
False
SETTINGS_MODULE	
'settings'
SITE_ID	
1
TEMPLATE_CONTEXT_PROCESSORS	
('django.core.context_processors.debug', 'django.core.context_processors.i18n')
TEMPLATE_DEBUG	
True
TEMPLATE_DIRS	
('/base/data/home/apps/s~rebuildingtogethercaptain-hrd/issue233.394576404661038179/templates', '/base/data/home/apps/s~rebuildingtogethercaptain-hrd/issue233.394576404661038179/room/templates', '/base/data/home/apps/s~rebuildingtogethercaptain-hrd/issue233.394576404661038179/staff/templates')
TEMPLATE_LOADERS	
('django.template.loaders.filesystem.load_template_source', 'django.template.loaders.app_directories.load_template_source')
TEMPLATE_STRING_IF_INVALID	
''
TEST_DATABASE_CHARSET	
None
TEST_DATABASE_COLLATION	
None
TEST_DATABASE_NAME	
None
TEST_RUNNER	
'django.test.simple.run_tests'
TIME_FORMAT	
'P'
TIME_ZONE	
'UTC'
TRANSACTIONS_MANAGED	
False
URL_VALIDATOR_USER_AGENT	
'Django/1.0.2 final (http://www.djangoproject.com)'
USE_ETAGS	
False
USE_I18N	
True
YEAR_MONTH_FORMAT	
'F Y'
You're seeing this error because you have DEBUG = True in your Django settings file. Change that to False, and Django will display a standard 500 page.

**Comment by babybunny** [comment]
Easy to reproduce.  Looks like an erroneous value crept in there somehow. 

local variables:
sc	
[SiteCaptain(**{'captain': datastore_types.Key.from_path(u'Captain', 5542591844057088L, _app=u's~rebuildingtogethercaptain-hrd'), 'type': None, 'site': datastore_types.Key.from_path(u'NewSite', 5776698364657664L, _app=u's~rebuildingtogethercaptain-hrd')})]

**Comment by babybunny** [comment]
there are two of these in the datastore, both on site with key 5776698364657664

both captains on this site have type=None.  They are Chris and Rick, BTW.
http://rebuildingtogethercaptain.appspot.com/room/site/list/5776698364657664/

If these captains' types get fixed manually the export should work. 

I was able to add a captain with type=None from the site edit page.  
http://rebuildingtogethercaptain.appspot.com/room/site/edit/5776698364657664/

this is a bug and I've created issue #241 to track it. 

Meanwhile I've removed these captains from the site and the export is no longer blocked on this error. now it's blocked on a new error.  still troubleshooting.
